src/features/dataset/DatasetHeaderLayout.tsx

```diff
@@ -28,25 +28,28 @@ export interface DatasetHeaderLayoutProps {
   headerLoading?: boolean
   // whether to show a bottom border
   border?: boolean
-  // editable reflects whether the user owns this dataset and thus can edit the name and title
-  editable?: boolean
-  // affects the behavior of the title. if true, the user can edit the title inline
-  manualCreation?: boolean
+  // allows editing the name, which will call onNameChange
+  nameEditable?: boolean
   // fires when the user renames the dataset
-  onRename?: (_: string, value: string) => void
+  onNameChange?: (_: string, value: string) => void
+  // allows editing the title, which will call onTitleChange
+  titleEditable?: boolean
   // fires when the user changes the dataset title
   onTitleChange?: (_: string, value: string) => void
+  // whether the user owns the dataset
+  userOwnsDataset?: boolean
```

Comment:
I like this naming! Wouldn't be opposed to changing `editable` to `userOwnsDataset` in other places that are appropriate (not necessary for this PR, just making a note as we move forward)